### PR TITLE
convert fixture providers to classes

### PR DESCRIPTION
**Pull request by snopoke**
The main object here was to have the fixture provider ID co-located with the fixture provider.

Mostly just copy paste.

@TylerSheffels

### CODE REVIEW

File: corehq/apps/fixtures/fixturegenerators.py [hunk 1]
```diff
@@ -35,45 +35,51 @@ def item_lists_by_domain(domain):
...
+        items_by_type = defaultdict(list)
+
+        def _set_cached_type(item, data_type):
+            # set the cached version used by the object so that it doesn't
+            # have to do another db trip later
+            item._data_type = data_type
+
+        for global_fixture in global_types.values():
+            items = list(FixtureDataItem.by_data_type(user.domain, global_fixture))
+            _ = [_set_cached_type(item, global_fixture) for item in items]
```
Comment (dimagimon) [inline corehq/apps/fixtures/fixturegenerators.py]: F841 local variable '_' is assigned to but never used

File: custom/m4change/fixtures/location_fixtures.py [hunk 2]
```diff
@@ -39,21 +34,18 @@ def to_fixture(self):
...
 
         locations_element = ElementTree.Element('locations')
-        locations = []
-        locations.append(Location.get(self.location_id))
-        for location in locations:
-            location_element = ElementTree.Element('location', attrib={
-                'name': location.name,
-                'id': location.get_id
-            })
-            locations_element.append(location_element)
```
Comment (snopoke) [inline custom/m4change/fixtures/location_fixtures.py]: Pointless to iterate over a loop with 1 item in it which we just added.

File: corehq/apps/fixtures/fixturegenerators.py (hunk 1 above)
Comment (czue) [inline corehq/apps/fixtures/fixturegenerators.py]: might be a place where `map(lambda item: _set_cached_type(item, global_fixture), items)` is better than a list comp

Comment (czue) [inline corehq/apps/fixtures/fixturegenerators.py]: (though i see this just moved from somewhere)

**Comment by czue** [comment]
kicked build - no blockers from me when tests pass but didn't review too closely. looks like a good change!

**Comment by snopoke** [comment]
:white_check_mark: build passing

**Pull request merged**